Ladybird/Qt/BrowserWindow.cpp

```diff
@@ -657,6 +685,34 @@ void BrowserWindow::tab_favicon_changed(int index, QIcon const& icon)
     m_tabs_container->setTabIcon(index, icon);
 }
 
+template<typename ClickHandler>
+static QToolButton* create_tab_button(BrowserWindow* window, QIcon icon, ClickHandler&& handler)
+{
+    auto* button = new QToolButton();
+    button->setIcon(icon);
+    button->setAutoRaise(true);
+    button->resize({ 20, 20 });
+
+    window->connect(button, &QPushButton::clicked, window, forward<ClickHandler>(handler));
+
+    return button;
+}
+
+void BrowserWindow::create_close_button_for_tab(Tab* tab)
```

Comment:
nit: This creates the audio button too, so maybe call it `create_buttons_for_tab()`?

Reply:
It creates a single button - I'd only pluralize it if it returned e.g. a list of buttons. Moot anyways, this function is actually going away due to macOS changes.